### PR TITLE
Feature/feature remote device

### PR DESCRIPTION
**Pull request by hhou2011**
@Pacane @d-shapiro

### CODE REVIEW

**Review by d-shapiro — CHANGES_REQUESTED** [review]
restoreLastSession is still bugged - devices and points don't get restored. Their nodes are there, but without actions.

File: src/main/java/org/dsa/iot/knx/KnxIPConnection.java [hunk 1]
```diff
@@ -345,8 +346,9 @@ void updatePointValue(GroupAddress address, byte[] asdu) {
 		if (null != point) {
 			Value value = null;
 			DatapointType type = point.getType();
-
-			DPT dpt = type.dpt;
+			String unit = null;
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/KnxIPConnection.java]: unit can be declared inline

File: src/main/java/org/dsa/iot/knx/KnxIPConnection.java [hunk 2]
```diff
@@ -586,8 +596,10 @@ public void readPoint(String group, EditablePoint point) {
 		Float valFloat = null;
 		Boolean valBoolean = null;
 		String valString = null;
+		String unit = null;
 
-		DPT dpt = type.dpt;
+		DPT dpt = type.getDpt();
+		unit = dpt.getUnit();
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/KnxIPConnection.java]: unit can be inline

File: src/main/java/org/dsa/iot/knx/datapoint/DPT1BitControlled.java [hunk 3]
```diff
@@ -16,4 +16,15 @@ public DPT1BitControlled(String dptId, String zeroLabel, String oneLabel) {
...
 	}
+
+	@Override
+	public String getLabel(boolean isZero) {
+		return isZero ? this.zeroLabel : this.oneLabel;
+	}
+
+	@Override
+	public String getUnit() {
+		// TODO Auto-generated method stub
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/datapoint/DPT1BitControlled.java]: missing implementation

File: src/main/java/org/dsa/iot/knx/datapoint/DPT2ByteFloat.java [hunk 4]
```diff
@@ -17,4 +17,15 @@ public int getDataTypeId() {
 		return DataTypes.NUMERIC.ordinal();
 	}
 
+	@Override
+	public String getLabel(boolean isZero) {
+		// TODO Auto-generated method stub
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/datapoint/DPT2ByteFloat.java]: missing implementation

File: src/main/java/org/dsa/iot/knx/datapoint/DPT2ByteUnsigned.java [hunk 5]
```diff
@@ -16,4 +16,15 @@ public DPT2ByteUnsigned(String dptId, String format, String suffix) {
 	public int getDataTypeId() {
 		return DataTypes.NUMERIC.ordinal();
 	}
+
+	@Override
+	public String getLabel(boolean isZero) {
+		// TODO Auto-generated method stub
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/datapoint/DPT2ByteUnsigned.java]: missing implementation

File: src/main/java/org/dsa/iot/knx/datapoint/DPT3BitControlled.java [hunk 6]
```diff
@@ -16,4 +16,15 @@ public DPT3BitControlled(String dptId, String format, String suffix) {
 	public int getDataTypeId() {
 		return DataTypes.NUMERIC.ordinal();
 	}
+
+	@Override
+	public String getLabel(boolean isZero) {
+		// TODO Auto-generated method stub
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/datapoint/DPT3BitControlled.java]: missing implementation

File: src/main/java/org/dsa/iot/knx/datapoint/DPT4ByteFloat.java [hunk 7]
```diff
@@ -16,4 +16,15 @@ public DPT4ByteFloat(String dptId, String format, String suffix) {
 	public int getDataTypeId() {
 		return DataTypes.NUMERIC.ordinal();
 	}
+
+	@Override
+	public String getLabel(boolean isZero) {
+		// TODO Auto-generated method stub
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/datapoint/DPT4ByteFloat.java]: missing implementation

**Review by Pacane — CHANGES_REQUESTED** [review]
multiple missing implementations for getLabel?

File: src/main/java/org/dsa/iot/knx/datapoint/DPT1BitControlled.java (hunk 3 above)
Comment (hhou2011) [reply src/main/java/org/dsa/iot/knx/datapoint/DPT1BitControlled.java]: For the controlled bit, no unit, but the label.

File: src/main/java/org/dsa/iot/knx/datapoint/DPT2ByteFloat.java (hunk 4 above)
Comment (hhou2011) [reply src/main/java/org/dsa/iot/knx/datapoint/DPT2ByteFloat.java]: For the number type, just the unit, no label.

**Review by d-shapiro — APPROVED** [review]

**Review by Pacane — APPROVED** [review]

**Pull request merged**